src/theme/components/hero.ts

```diff
@@ -189,12 +189,18 @@ const campaign = {
     maxWidth: { md: "1248px" },
     position: { md: "relative" },
     zIndex: 2,
+    ".chakra-heading": {
+      color: "ui.typography.inverse.heading",
+    },
     _dark: {
       color: "dark.ui.typography.body",
+      ".chakra-heading": {
+        color: "dark.ui.typography.heading",
+      },
```

Comment:
@EdwinGuzman This is where I am adding styles to set the color of all headings within the campaign hero.  It sets the color, as expected, but it also removes the ability to set custom colors for the headings.  If you look in the stories file, you'll see that I have custom text values all over the place for the campaign heroes, but none of the custom colors are showing.  This solution would be a problem when, like in DC facelift (DCF), the background color is set to a light color and the text color then needs to be set to a dark color.

If we don't set the color in the component styles, then you are able to set custom colors for the heading.  And again, this is what they are doing successfully in DCF.  However, this falls apart when dark mode is enabled.  I know DCF currectly is not offering dark mode, but they will some day.  Additionally, DCF will not be the only site where this could be a problem.

If you have a chance, can you see if you cam find a way around this?  Or at least take a look and then let's talk about it.  I don't think we are going to solve the heading color issue with this type of solution, but I do have some ideas for how we can resolve this.

Reply:
This is quite the issue. I don't think DCF is doing it successfully because they have to add `!important` for the colors to show. I did the same by doing 

```
__css={{ color: "magenta !important", _dark: { color: "yellow" } }}
```

and this worked. My first thought was to do more of what the `Hero` component already does with styles similar to the `campaignBgStyles` or `backgroundImageStyle` css objects. So we would do more preprocessing in the component for styles to pass. Another option is to try to clone and set the proper styles in the component similar to this:

```
const finalHeading =
  heading && React.cloneElement(heading, { __css: headingStyles });
```

But none of these approaches work. We are assuming devs will pass DS `Heading` components and it's also too complicated for the `Hero` component to parse the `subHeaderText` prop to find and update any `Heading` components it finds.

What if we don't set any default styles or loosely set them so they are very easy to override? Then the onus really falls on the consuming app to implement the `Hero` component correctly. Theoretically, to me, I think this is okay because we can ask designers to check for color contrast and proper color values at the design stage before a dev implements it.

We should talk about this next week. I'm curious about your ideas to resolve this.

Reply:
@EdwinGuzman 
Ultimately, the color of the text, especially in the `Hero` component, shoudl be the default text colors for light and dark mode or the inverse text colors for light and dark mode.  I feel strongly about this.  If this is what we want, then we can set the styles as I've done, which will force the colors to the default light and dark mode text colors.  We can also add a new `isInverseTextColor` prop.  And if `isInverseTextColor` is true, the component will use the `inverse` text colors.  Then, as you mentioned, it is up to the designer to specify which color should be used: `default` or `inverse`.

Reply:
If we make this change, would it mean that any apps currently using the component will need an update when this update is deployed?

Reply:
> If we make this change, would it mean that any apps currently using the component will need an update when this update is deployed?

I don't think so.  Searching the code, I only see DC and MLN using the DS campaign hero.

https://github.com/search?q=org%3ANYPL+heroType%3D%22campaign%22+-repo%3ANYPL%2Fnypl-design-system+-repo%3ANYPL%2Fnypl-ds-test-app&type=code